build/gulp/tasks/perf.ts

```diff
@@ -83,25 +83,28 @@ task('perf:build', cb => {
 task('perf:run', async () => {
   const measures: ProfilerMeasureCycle[] = []
   const times = argv.times || DEFAULT_RUN_TIMES
+  let browser
 
-  const browser = await puppeteer.launch()
+  try {
+    browser = await puppeteer.launch()
 
-  for (let i = 0; i < times; i++) {
-    const page = await browser.newPage()
-
-    try {
+    for (let i = 0; i < times; i++) {
+      const page = await browser.newPage()
       await page.goto(`http://${config.server_host}:${config.perf_port}`)
 
       const measuresFromStep = await page.evaluate(() => window.runMeasures())
       measures.push(measuresFromStep)
-    } catch (e) {
-      throw e
```

Comment:
as we are just rethrowing here, `try - finally` should be enough. The same applies to the `browser` as well.

In future we might think to encapsulate it into abstraction like
```js
withResource(browser, async () => {
   ... logic that might throw ..
}) // resource is guaranteed to be released here
```